sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_deserialize.py

```diff
@@ -69,56 +78,58 @@ def deserialize_file_properties(response, obj, headers):
     return file_properties
 
 
-def deserialize_path_properties(path_list):
-    return [PathProperties._from_generated(path) for path in path_list] # pylint: disable=protected-access
+def deserialize_path_properties(path_list: List["Path"]) -> List[PathProperties]:
+    return [PathProperties._from_generated(path) for path in path_list]  # pylint: disable=protected-access
 
 
-def return_headers_and_deserialized_path_list(response, deserialized, response_headers):  # pylint: disable=name-too-long, unused-argument
+def return_headers_and_deserialized_path_list(  # pylint: disable=name-too-long, unused-argument
+    _,
+    deserialized: "PathList",
+    response_headers: Dict[str, Any]
+) -> Tuple[Collection["Path"], Dict[str, Any]]:
     return deserialized.paths if deserialized.paths else {}, normalize_headers(response_headers)
 
 
-def get_deleted_path_properties_from_generated_code(generated):  # pylint: disable=name-too-long
+def get_deleted_path_properties_from_generated_code(generated: "BlobItemInternal") -> DeletedPathProperties:  # pylint: disable=name-too-long
     deleted_path = DeletedPathProperties()
     deleted_path.name = generated.name
-    deleted_path.deleted_time = generated.properties.deleted_time
-    deleted_path.remaining_retention_days = generated.properties.remaining_retention_days
-    deleted_path.deletion_id = generated.deletion_id
+    deleted_path.deleted_time = generated.properties.deleted_time  # type: ignore [assignment]
```

Comment:
Can't directly change blob types or `DeletedPathProperties` since the existing non-optional types are needed for other methods. The best I can think of as of now is type ignore. Any suggestions?